platform/test/spec/testToolManagerSpec.js

```diff
@@ -1,18 +1,185 @@
+
+            // Check the expected results
+            for (const t in tm.toolsUrls){
+                expect(tm.toolsUrls[t].url).toEqual(TOOL_URLS[t]);
+            }
+        })
+
+        it("the tool configuration is fetched from the remote given by its URL", () => {
+            // Call the target object
+            tm.setToolsUrls(TOOL_URLS);
+
+            // Check the expected results
+            expect(XMLHttpRequest.prototype.open).toHaveBeenCalledWith("GET", TOOL_URLS[0],false);
+        })
+
+        it("the tool configuration is parsed and stored", () => {
+            jasmine.Ajax.stubRequest('test://t1.url/tool-config.json').andReturn({
+                "responseText": TOOL_1PANELDEF_1FUNCTION,
+                "status": 200
+            });
+
+            // Call the target object
+            tm.setToolsUrls(TOOL_URLS);
+
+            // Check the expected results
+            const TOOL1_ID= "tool-1"
+
+            expect(tm.tools[TOOL1_ID]).toBeInstanceOf(Object);
+        })
+
+        it("the tool configuration that is parsed and stored has the expected keys", () => {
+            jasmine.Ajax.stubRequest('test://t1.url/tool-config.json').andReturn({
+                "responseText": TOOL_1PANELDEF_1FUNCTION,
+                "status": 200
+            });
+
+            // Call the target object
+            tm.setToolsUrls(TOOL_URLS);
+
+            // Check the expected results
+            const TOOL1_ID= "tool-1"
+            const EXPECTED_TOOL_KEYS = ["id", "name","version", "author", "homepage", "functions", "panelDefs"];
+
+            const storedToolKeys = Object.keys(tm.tools[TOOL1_ID]);
+            for (let key of EXPECTED_TOOL_KEYS){
+                expect( storedToolKeys.find(n => n===key ) ).toEqual(key);
+            }
+        })
+    })
+
+    describe("panel definitions can be retrieved", () => {
+        // Setup
+        const TOOL_URLS= [ "test://t1.url/tool-config.json" ];
+        const PANEL_DEF_ID = "paneldef-t1";
+        let tm;
+
+        beforeEach( () => {
+            jasmine.Ajax.install();
+           
+            spyOn(XMLHttpRequest.prototype, "open").and.callThrough();
+            spyOn(XMLHttpRequest.prototype, "send").and.callThrough();
+
+            tm = new ToolManager();
+        })
+
+        afterEach(function() {
+            jasmine.Ajax.uninstall();
+        });  
+
+        it ("for a panel definition id that exists its object is returned", () => {
+            jasmine.Ajax.stubRequest('test://t1.url/tool-config.json').andReturn({
+                "responseText": TOOL_1PANELDEF_1FUNCTION,
+                "status": 200
+            });
+            tm.setToolsUrls(TOOL_URLS);
+
+            // Call the target object
+            const foundPanelDef= tm.getPanelDefinition(PANEL_DEF_ID);
+
+            // Check the expected results
+            expect(foundPanelDef).toBeInstanceOf(Object);
+            expect(foundPanelDef.id).toEqual(PANEL_DEF_ID);
```

Comment:
Check more of the panel-definition contents?

Reply:
Agree, updated test to check the complete panel definition object rather than just the id. Fixed in 0f8cd9c894cfa9fc709b55c649f8af02bfd66cf5.